core/domain/caching_services_test.py

```diff
@@ -936,14 +936,19 @@ def test_platform_parameters_with_unicode_are_set_and_get_correctly(
                 [platform_parameter_id]),
             {})
 
-        default_parameter = parameter_domain.PlatformParameter.from_dict({ # type: ignore[no-untyped-call]
+        default_parameter = parameter_domain.PlatformParameter.from_dict({
             'name': 'parameter_a 😍',
             'description': '😍😍😍😍',
             'data_type': 'bool',
             'rules': [
                 {
                     'filters': [
-                        {'type': 'server_mode', 'conditions': [['=', 'prod']]}],
+                        # The expected type of 'conditions' key is
+                        # List[Tuple[str, str]], but for testing purposes
+                        # we are providing value of type List[List[str]].
+                        # Which causes MyPy to throw error, Thus to silent
+                        # the error we used ignore here.
+                        {'type': 'server_mode', 'conditions': [['=', 'prod']]}],  # type: ignore[list-item]
```

Comment:
If we change `conditions` to `List[Tuple[str, str]]` (`[('=', 'prod')]`) then the backend tests fail, because for testing purposes this dictionary is firstly stored in backend and fetch again, and then this dictionary is compared with the fetched dictionary. In fetched dictionary, `conditions` is always returned as a `[['=', 'prod']]` no matter if we change the input dictionary's `conditions` to `List[Tuple[str, str]]` (`[('=', 'prod')]`)  or not. So, to keep passing the test I did not change `conditions`.

Reply:
Oh makes sense, so it must be saved in the DB as a `JsonProperty` and tuples would be converted to lists.

Reply:
If we change the fetched data still we have to add `ignore:` in tests. So to skip the ignores i changed the tuple to lists.

Reply:
Basically i changed the type of `conditions` from tuple to list so that we don't have to use ignores anymore